app/controllers/api/enterprise_attachment_controller.rb

```diff
@@ -0,0 +1,37 @@
+module Api
+  class EnterpriseAttachmentController < BaseController
+    class MissingImplementationError < StandardError; end
+    class UnknownEnterpriseAuthorizationActionError < StandardError; end
+
+    before_filter :load_enterprise
+
+    respond_to :json
+
+    def destroy
+      return respond_with_conflict(error: destroy_attachment_does_not_exist_error_message) unless @enterprise.send("#{attachment_name}?")
+
+      @enterprise.update_attributes!(attachment_name => nil)
+      render json: @enterprise, serializer: Api::Admin::EnterpriseSerializer, spree_current_user: spree_current_user
```

Comment:
being picky here, but the `Api` namespace is redundant here since this code is defined within `module Api`.